### ISSUE TITLE
yolo v3 region 106 NAN while others are numbers

### ISSUE THREAD

**Issue by charlesLucky** [AlexeyAB/darknet#3016]
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000001, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.748791, Class: 0.955784, Obj: 0.939010, No Obj: 0.001280, .5R: 1.000000, .75R: 0.666667,  count: 3
Region 94 Avg IOU: 0.495121, Class: 0.923221, Obj: 0.872888, No Obj: 0.000077, .5R: 0.000000, .75R: 0.000000,  count: 1
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000002, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.828836, Class: 0.984393, Obj: 0.756534, No Obj: 0.001503, .5R: 1.000000, .75R: 1.000000,  count: 4
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000004, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000002, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.774369, Class: 0.956332, Obj: 0.293226, No Obj: 0.000885, .5R: 1.000000, .75R: 0.666667,  count: 3
Region 94 Avg IOU: 0.722639, Class: 0.962963, Obj: 0.935729, No Obj: 0.000093, .5R: 1.000000, .75R: 0.000000,  count: 1
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000002, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.747990, Class: 0.988499, Obj: 0.676930, No Obj: 0.001658, .5R: 1.000000, .75R: 0.500000,  count: 4
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000004, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000002, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.721379, Class: 0.992091, Obj: 0.972022, No Obj: 0.001113, .5R: 1.000000, .75R: 0.333333,  count: 3
Region 94 Avg IOU: 0.532948, Class: 0.979501, Obj: 0.673004, No Obj: 0.000073, .5R: 1.000000, .75R: 0.000000,  count: 1
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000002, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.848636, Class: 0.989794, Obj: 0.720877, No Obj: 0.001066, .5R: 1.000000, .75R: 1.000000,  count: 3
Region 94 Avg IOU: 0.217985, Class: 0.927282, Obj: 0.023733, No Obj: 0.000095, .5R: 0.000000, .75R: 0.000000,  count: 1
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000001, .5R: -nan, .75R: -nan,  count: 0
1267: 0.246899, 0.262875 avg, 0.001000 rate, 3.298116 seconds, 81088 images

Dear friend

when I train my model, I get the above logs, could anyone help me, why I get NAN for Region 106  and Region 94??? 

My object is not small, it is 76*100 if resize the image to 416*416.

here is my cfg:

# Training
batch=64
subdivisions=16
width=416    #608 32
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
.........
[convolutional]
size=1
stride=1
pad=1
filters=39
activation=linear

[yolo]
mask = 0,1,2
anchors = 10,13,  16,30,  33,23,  30,61,  62,45,  59,119,  116,90,  156,198,  373,326
classes=8
num=9
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1

**Comment by AlexeyAB** [comment]
> when I train my model, I get the above logs, could anyone help me, why I get NAN for Region 106 and Region 94???

This is normal. When there are no objects for layers 94, 106 then there are NAN values.
https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects

> Note: If during training you see nan values for avg (loss) field - then training goes wrong, but if nan is in some other lines - then training goes well.

**Comment by charlesLucky** [comment]
> > when I train my model, I get the above logs, could anyone help me, why I get NAN for Region 106 and Region 94???
> 
> This is normal. When there are no objects for layers 94, 106 then there are NAN values.
> https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects
> 
> > Note: If during training you see nan values for avg (loss) field - then training goes wrong, but if nan is in some other lines - then training goes well.

@AlexeyAB  thanks so much for your explain, in my log I find all the region 106 are nan, and when I use the weights file to test the detection, it didn't work, so I am wondering if I can only use the region 82 or 94 to do the detection? If so, how can I do it?
Below is a part of my log, it seems layer 82 and 94 get good accuracy:

Region 82 Avg IOU: 0.776366, Class: 0.996601, Obj: 0.996942, No Obj: 0.001867, .5R: 1.000000, .75R: 0.500000,  count: 2
Region 94 Avg IOU: 0.733412, Class: 0.962309, Obj: 0.231898, No Obj: 0.000062, .5R: 1.000000, .75R: 0.000000,  count: 2
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000001, .5R: -nan, .75R: -nan,  count: 0

**Comment by charlesLucky** [comment]
seems it works even though layer 106 gets NAN
